tests/suites/test_suite_nist_kw.function

```diff
@@ -158,19 +158,17 @@ void nist_kw_plaintext_lengths( int in_len, int out_len, int mode, int res )
 
     memset( key, 0, sizeof( key ) );
 
-    if (in_len == 0)
+    if( in_len != 0 )
     {
-        /* mbedtls_calloc can return NULL for zero-length buffers. Make sure we
-         * always have a plaintext buffer, even if the length is 0. */
-        plaintext = mbedtls_calloc( 1, 1 );
+        plaintext = mbedtls_calloc( 1, in_len );
     }
-    else
+    TEST_ASSERT( in_len == 0 || plaintext != NULL );
```

Comment:
Move this inside the `if( if_len != 0 )` and simplify it to `TEST_ASSERT( plaintext != NULL )`.